Replace mapfile usage with POSIX-compatible array loading

rhcp011235 opened the pull request:
macOS ships an old Bash (3.2) which doesn’t support mapfile/readarray. This was causing the script to fail on systems that aren’t using Homebrew’s Bash. Swapped the mapfile block for a portable while-read loop so the script now runs correctly on stock macOS bash without requiring any shell upgrades.

rhcp011235 committed:
Replace mapfile usage with POSIX-compatible array loading

macOS ships an old Bash (3.2) which doesn’t support mapfile/readarray. This was causing the script to fail on systems that aren’t using Homebrew’s Bash. Swapped the mapfile block for a portable while-read loop so the script now runs correctly on stock macOS bash without requiring any shell upgrades.

PoomSmart commented:
1. Don't any other shell scripts need update to make them compatible with old Bash?
2. The file header comment block is unnecessary because if you do add that, other files also need that added for consistency
3. README isn't updated to remove Bash 5 requirement, and therefore this PR is incomplete at best
4. See concerning reviews from Copilot above

rhcp011235 commented:
> 1. Don't any other shell scripts need update to make them compatible with old Bash?
> 2. The file header comment block is unnecessary because if you do add that, other files also need that added for consistency
> 3. README isn't updated to remove Bash 5 requirement, and therefore this PR is incomplete at best
> 4. See concerning reviews from Copilot above

I dont use AI, I am a real programmer. ive only modified what I showed you, not modified anything else. this works on the default zsh shell on Mac OS. Im on a M4 max MacBook Pro Darwin M4-Macbook-Pro.localdomain 25.1.0 Darwin Kernel Version 25.1.0: Mon Oct 20 19:34:05 PDT 2025; root:xnu-12377.41.6~2/RELEASE_ARM64_T6041 arm64

╭─    ~/MGKeys  on   master ──────

rhcp011235 commented:
here is output: 

[INFO] Successfully extracted dyld_shared_cache from IPSW
[INFO] Found cache directory: dyld_shared_cache/23B85__iPhone18,2
[INFO] Extracting libMobileGestalt.dylib from cache...
[INFO] Successfully extracted: dyld_shared_cache/23B85__iPhone18,2/libMobileGestalt.dylib
[INFO] Using dylib: libMobileGestalt.dylib
[INFO] Running key discovery...
Using architecture: arm64e
Warning: nm tools failed, using strings extraction (may include false positives)
Extracted 826 symbols
Processed 570 hashes
Found 172 maybe-non-gestalt keys
Output written to maybe-non-gestalt-keys.txt
Discovery complete!
Results:
  - Readable keys: readable.txt
  - Obfuscated mapped: discover-obfuscated-mapped.txt
  - Maybe non-gestalt: maybe-non-gestalt-keys.txt
Warning: arch-specific strings failed, using plain strings
Extracted 1223 potential hashes
Wrote 1174 hashes to hashes.txt
Warning: 1XU1vjgruwDJdBGH+Gx9lg not found in hashes.txt
Warning: aYjxnADoQ2HXS8J+swzuNg not found in hashes.txt
Warning: E+9+cggZ2rXU0z8mWiOung not found in hashes.txt
Warning: jtg9JUxH6Mf8k7m7j8omug not found in hashes.txt
Warning: ZfUfxSFklLDVCKbgOE7kpg not found in hashes.txt
Warning: 1XU1vjgruwDJdBGH+Gx9lg not found in hashes.txt
Warning: aYjxnADoQ2HXS8J+swzuNg not found in hashes.txt
Warning: E+9+cggZ2rXU0z8mWiOung not found in hashes.txt
Warning: jtg9JUxH6Mf8k7m7j8omug not found in hashes.txt
Warning: ZfUfxSFklLDVCKbgOE7kpg not found in hashes.txt
[INFO] Discovery complete!
❯ cat readable.txt
AbsoluteDepthLimit
AccelerometerCapability
AccessibilityCapability
AcousticID
ActivationProtocol
ActiveWirelessTechnology
ActuatorResonantFrequency
AdditionalTextTonesCapability
ADSCapability
AggregateDevicePhotoZoomFactor

rhcp011235 commented:
PS: also some bug with nm tools as well. working on seeing how to fix this. seems like Mac wants it to run via xcrun llvm-nm

PoomSmart commented:
This is not about using AI or not. I'm sure you tested it. However, there are 4 items above that are expected to be addressed.

rhcp011235 commented:
> This is not about using AI or not. I'm sure you tested it. However, there are 4 items above that are expected to be addressed.

lot more than 4. so get fixing, i did my part to get it working on my machine. this is what is wrong with VIBE coded stuff.

PoomSmart commented:
No sign of addressing concerns from the project owner.

> lot more than 4. So get fixing

Toxic.

This conversation is going nowhere.
